### PR TITLE
created function and tests for shrink()

### PR DESCRIPTION
**Pull request by rita-ni**

### CODE REVIEW

File: tests/testthat/test-shrink.R [hunk 1]
```diff
@@ -1,5 +1,24 @@
-test_that("shrink works", {
-  old_img <- array(1:24, dim = c(3, 4, 2))
 
-  expect_identical(sharpen(old_img), old_img)
+image <- array(1:24, dim = c(50, 50, 3))
```
Comment (franklu2014) [inline tests/testthat/test-shrink.R:2]: The dimension is off; this array should throw an error.

**Review by franklu2014 — CHANGES_REQUESTED** [review]

File: tests/testthat/test-shrink.R (hunk 1 above)
Comment (franklu2014) [reply tests/testthat/test-shrink.R:2]: I misunderstood how `array()` works; should be fine

File: R/shrink.R [hunk 2]
```diff
@@ -8,9 +9,48 @@
...
+  crop_col <- dim(image)[2] - width
+
+  if (crop_row %% 2 == 0){
+    top_row <- as.integer(crop_row/2)
+    bottom_row <- as.integer(dim(image)[1] - top_row)
+  }
+  else {
+    top_row <- as.integer((crop_row + 1)/2)
+    bottom_row <- as.integer(dim(image)[1] - top_row + 1)
+  }
```
Comment (franklu2014) [inline R/shrink.R:41]: Minor question and no need for change: not sure if I understand you right, but does the code below equivalent to yours?
```suggestion
top_row <- crop_row %/% 2
bottom_row <- dim(image)[1] - (crop_row - top_row)
```

File: R/shrink.R [hunk 3]
```diff
@@ -8,9 +9,48 @@
...
+
+  if (crop_col %% 2 == 0){
+    left_col <- as.integer(crop_col/2)
+    right_col <- as.integer(dim(image)[2] - left_col)
+  }
+  else {
+    left_col <- as.integer((crop_col + 1)/2)
+    right_col <- as.integer(dim(image)[2] - left_col + 1)
+  }
```
Comment (franklu2014) [inline R/shrink.R:50]: Similar minor question and no need for change: is the code below also equivalent:
```suggestion
left_col <- crop_col %/% 2
right_col <- dim(image)[2] - (crop_col - left_col)
```

File: R/shrink.R (hunk 2 above)
Comment (rita-ni) [reply R/shrink.R:41]: Yes I think both codes are legit and the result should be same!

File: R/shrink.R (hunk 3 above)
Comment (rita-ni) [reply R/shrink.R:50]: Correct, they are doing the same thing in essence, your solution is more neat and fancier. But I find it more straightforward to compute it the more basic way, thanks for bringing this up, i didn't know about the %/% operator before!

Comment (franklu2014) [reply R/shrink.R:50]: yeah, they essentially do the same thing.  I just googled that `%/%` thing haha.  will merge now.

**Review by franklu2014 — APPROVED** [review]

**Pull request merged**